pkg/provisioner/provisioner.go

```diff
@@ -88,6 +88,12 @@ type Provisioner interface {
 	// IsReady checks if the provisioning backend is available to accept
 	// all the incoming requests.
 	IsReady() (result bool, err error)
+
+	// IsProvisioning verifies if the current host is currently being provisioned or not
+	IsProvisioning() bool
+
+	// CanBeProvisioned checks if the backend has a free provisioning slot
+	CanBeProvisioned() bool
```

Comment:
Couldn't these be collapsed into a single HasProvisioningCapacity() function? They already share state.

Reply:
I reviewed all the checks currently applied, and it seems ok to merge them together in all the three cases (at the beginning of the state machine and when transitioning). Done